### PR TITLE
Add `MultiTurnSample` conversion path for RAGAS trajectory-aware agent metrics

### PR DESCRIPTION
**Pull request by yczhang-nv**
## Description

The RAGAS ATIF adapter only converted to `SingleTurnSample`, limiting evaluation to metrics like `AnswerAccuracy`. Trajectory-aware metrics (`AgentGoalAccuracyWithoutReference`, `ToolCallAccuracy`) require `MultiTurnSample` and previously failed with a validation error.

This PR adds a `MultiTurnSample` conversion path alongside the existing `SingleTurnSample` path, controlled by a new `sample_type` config field (`single_turn` | `multi_turn`).

Closes

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/NeMo-Agent-Toolkit/blob/develop/docs/source/resources/contributing/index.md).
- We require that all contributors "sign-off" on their commits. This certifies that the contribution is your original work, or you have rights to submit it under the same license, or a compatible license.
  - Any contribution which contains commits that are not Signed-Off will not be accepted.
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive guidance on multi-turn Ragas metrics for evaluating agent workflows with tool calls, including configuration examples and metric recommendations.

* **New Features**
  * Introduced multi-turn evaluation mode via new `sample_type` configuration option (defaults to single-turn for backward compatibility).
  * Enabled assessment of tool call accuracy and agent goal metrics in multi-turn agent conversations.

* **Tests**
  * Added comprehensive test coverage for multi-turn evaluation scenarios and edge cases.

### CODE REVIEW

File: packages/nvidia_nat_ragas/src/nat/plugins/ragas/rag_evaluator/atif_evaluate.py [hunk 1]
```diff
@@ -53,32 +61,132 @@ def _trajectory_to_retrieved_contexts(trajectory: ATIFTrajectory) -> list[str]:
...
+
+
+def _atif_step_to_ragas_messages(step: ATIFStep, ) -> list["HumanMessage | AIMessage | ToolMessage"]:
+    """Convert a single ATIF step into one or more RAGAS message objects.
+
+    Mapping:
+        source="user"  → HumanMessage
+        source="agent" → AIMessage (with optional ToolCall list)
+                         followed by ToolMessage per observation result
+        source="system" → HumanMessage (best-effort; RAGAS has no SystemMessage)
```
Comment (willkill07) [inline packages/nvidia_nat_ragas/src/nat/plugins/ragas/rag_evaluator/atif_evaluate.py:75]: I believe this should fix the [documentation build failure](https://github.com/NVIDIA/NeMo-Agent-Toolkit/actions/runs/22879739483/job/66380111461?pr=1774#step:5:3079)
```suggestion
    Mapping:

    * source="user"  → HumanMessage
    * source="agent" → AIMessage (with optional ToolCall list)
                         followed by ToolMessage per observation result
    * source="system" → HumanMessage (best-effort; RAGAS has no SystemMessage)
```

**Review by willkill07 — APPROVED** [review]

**Comment by AnuradhaKaruppiah** [comment]
@yczhang-nv thx for making these changes, I will be taking this over for integration with the new ragas version
